src/EditorFeatures/CSharp/AutomaticCompletion/AutomaticLineEnderCommandHandler_Helpers.cs

```diff
@@ -0,0 +1,1025 @@
+                .WithOpenBraceToken(GetOpenBrace(editorOptions));
+
+        private static BlockSyntax GetBlockNode(IEditorOptions editorOptions)
+            => SyntaxFactory.Block().WithOpenBraceToken(GetOpenBrace(editorOptions)).WithCloseBraceToken(GetCloseBrace(editorOptions));
+
+        private static SyntaxToken GetOpenBrace(IEditorOptions editorOptions)
+            => SyntaxFactory.Token(
+                    leading: SyntaxTriviaList.Empty,
+                    kind: SyntaxKind.OpenBraceToken,
+                    trailing: SyntaxTriviaList.Create(GetNewLineTrivia(editorOptions)))
+                .WithAdditionalAnnotations(s_openBracePositionAnnotation);
+
+        private static SyntaxToken GetCloseBrace(IEditorOptions editorOptions)
+            => SyntaxFactory.Token(
+                leading: SyntaxTriviaList.Empty,
+                kind: SyntaxKind.CloseBraceToken,
+                trailing: SyntaxTriviaList.Create(GetNewLineTrivia(editorOptions)));
+
+        private static SyntaxTrivia GetNewLineTrivia(IEditorOptions editorOptions)
+        {
+            var newLineString = editorOptions.GetNewLineCharacter();
+            return SyntaxFactory.EndOfLine(newLineString);
+        }
+
+        /// <summary>
+        /// Add braces to the <param name="node"/>.
+        /// For FieldDeclaration and EventFieldDeclaration, it will change them to PropertyDeclaration and EventDeclaration
+        /// </summary>
+        private static SyntaxNode WithBraces(SyntaxNode node, IEditorOptions editorOptions)
+            => node switch
+            {
+                BaseTypeDeclarationSyntax baseTypeDeclarationNode => WithBracesForBaseTypeDeclaration(baseTypeDeclarationNode, editorOptions),
+                ObjectCreationExpressionSyntax objectCreationExpressionNode => GetObjectCreationExpressionWithInitializer(objectCreationExpressionNode, editorOptions),
+                FieldDeclarationSyntax fieldDeclarationNode when fieldDeclarationNode.Declaration.Variables.IsSingle()
+                    => ConvertFieldDeclarationToPropertyDeclaration(fieldDeclarationNode, editorOptions),
+                EventFieldDeclarationSyntax eventFieldDeclarationNode => ConvertEventFieldDeclarationToEventDeclaration(eventFieldDeclarationNode, editorOptions),
+                BaseMethodDeclarationSyntax baseMethodDeclarationNode => AddBlockToBaseMethodDeclaration(baseMethodDeclarationNode, editorOptions),
+                LocalFunctionStatementSyntax localFunctionStatementNode => AddBlockToLocalFunctionDeclaration(localFunctionStatementNode, editorOptions),
+                AccessorDeclarationSyntax accessorDeclarationNode => AddBlockToAccessorDeclaration(accessorDeclarationNode, editorOptions),
+                _ when node.IsEmbeddedStatementOwner() => AddBlockToEmbeddedStatementOwner(node, editorOptions),
+                _ => throw ExceptionUtilities.UnexpectedValue(node),
```

Comment:
so can we hit here for things like a fieldDeclarationNode with multiple variables?  what happens/controls that otherwise?

Reply:
It won't hit here because before calling to change the document, there is an check [here](https://github.com/Cosifne/roslyn/blob/a85c119b78c5544242247318d452c47a5679555d/src/EditorFeatures/CSharp/AutomaticCompletion/AutomaticLineEnderCommandHandler_Helpers.cs#L498)
And I also have two tests [test1](https://github.com/Cosifne/roslyn/blob/a85c119b78c5544242247318d452c47a5679555d/src/EditorFeatures/CSharpTest/AutomaticCompletion/AutomaticLineEnderTests.cs#L1474)  [test2](https://github.com/Cosifne/roslyn/blob/a85c119b78c5544242247318d452c47a5679555d/src/EditorFeatures/CSharpTest/AutomaticCompletion/AutomaticLineEnderTests.cs#L1489) that test both field and event field